### PR TITLE
Remove testoperationvisitor workaround now that bug is fixed.

### LINKED ISSUE #23819: IConditionalAccessInstanceOperation.Syntax is null for a conditional access of off an implicit receiver within a With statement, Language property throws NullReferenceException because of that
```
    Sub Test5(Of T As I1)(x as C1(Of T))
        With x.F2
            ?.Test()
        End With
    End Sub 
```

Affected unit-tests:
```
Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.ConditionalAccessTests.CodeGen_04
```

### PR DESCRIPTION
**Pull request by 333fred**
Test only change. Fixes https://github.com/dotnet/roslyn/issues/23819 by removing the test workaround, as the original bug is now fixed. Ran all tests with the IOperation hook locally and all tests passed.

### CODE REVIEW

**Comment by 333fred** [comment]
Ping for reviews

File: src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs [hunk 1]
```diff
@@ -44,26 +44,17 @@ public override void Visit(IOperation operation)
                 var type = operation.Type;
                 var constantValue = operation.ConstantValue;
 
-                // operation.Language can throw due to https://github.com/dotnet/roslyn/issues/23819
+                Assert.NotNull(syntax);
```
Comment (AlekseyTs) [inline src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: >Assert.NotNull(syntax); [](start = 16, length = 23)

Would it be possible to add similar assert to the other visitor we are using for tests?

**Review by mavasani — APPROVED** [review]

File: src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs [hunk 2]
```diff
@@ -44,26 +44,17 @@ public override void Visit(IOperation operation)
                 var type = operation.Type;
                 var constantValue = operation.ConstantValue;
 
-                // operation.Language can throw due to https://github.com/dotnet/roslyn/issues/23819
```
Comment (AlekseyTs) [inline src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: >https://github.com/dotnet/roslyn/issues/23819 [](start = 55, length = 45)

Would it make sense to add a unit-test asserting IOperation tree for the scenario described in the issue? Or do we have one already?

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 1) with some suggestions

File: src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs (hunk 2 above)
Comment (mavasani) [reply src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: @333fred If you are not planning to add such a unit test, I can do it as part of https://github.com/dotnet/roslyn/issues/23819.

Comment (mavasani) [reply src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: Ah, never mind, seems that issue is assigned to Fred as well. I will let you decide which PR to add test.

File: src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs (hunk 1 above)
Comment (333fred) [reply src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: This is the PR to fix that issue :)
I'll add a test to this PR.

---
In reply to: [197882035](https://github.com/dotnet/roslyn/pull/28084#discussion_r197882035) [](ancestors = 197882035)

File: src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs (hunk 2 above)
Comment (333fred) [reply src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs]: This is the PR to fix that issue :)
I'll add a test to this PR.

---
In reply to: [197884158](https://github.com/dotnet/roslyn/pull/28084#discussion_r197884158) [](ancestors = 197884158)

**Comment by 333fred** [comment]
@jaredpar for approval of test-only change.

**Comment by 333fred** [comment]
Test failures are:
* ClosedChannelExeption: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/2327/console
* VS Licensing Popup: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/2326/console
* Test failure (disabled by https://github.com/dotnet/roslyn/pull/28106): https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_spanish_unit32_prtest/2128

**Comment by 333fred** [comment]
@dotnet-bot retest windows_debug_spanish_unit32_prtest please

**Comment by jaredpar** [comment]
approved

**Comment by 333fred** [comment]
Merging with the integration test failures, as those are caused by licensing issues which are being tracked by dceng.

**Pull request merged**